index.test-d.ts

```diff
@@ -133,3 +133,32 @@ convertElement()
 convertElement(null)
 convertElement(undefined)
 expectError(convertElement<Article>())
+
+declare const node: unknown
+
+/* Type assertion */
+if (isElement(node)) {
+  expectType<Element>(node)
+}
+
+if (
+  isElement<Section>(
+    node,
+    (node): node is Section => 'tagName' in node && node.tagName === 'section'
+  )
+) {
+  expectType<Section>(node)
+}
+
+if (isElement(node, node => 'children' in node)) {
+  expectType<unknown>(node)
+} else {
+  expectType<unknown>(node)
```

Comment:
So this is because of TS limitations, right? We could be sure that `node` is an element in the if-branch, but it could also be an element in the else-branch?
Maybe good to document it?

Reply:
Yes, it can not be an assertion in this case because in `else` it could be also `Element`, that's why I added `if (isElement(node) && 'children' in node)` below.